packages/artillery-plugin-memory-inspector/test/index.spec.mjs

```diff
@@ -0,0 +1,205 @@
+      }
+    }
+  });
+
+  //Act: run the test and get report
+  const output =
+    await $`TEST_PORT=${currentPort} ../artillery/bin/run run ./test/fixtures/scenario.yml --overrides ${override} --output ./test/output.json`;
+  const report = JSON.parse(fs.readFileSync('./test/output.json'));
+
+  //Assert:
+  //plugin doesn't mess with existing before scenario handlers
+  t.ok(
+    output.stdout.includes(
+      'Hello from the Handler!',
+      'Issue with running existing Before Scenario Handler!'
+    )
+  );
+
+  //sanity check that it can reach server
+  t.ok(
+    report.aggregate.counters['http.codes.200'] > 0,
+    'Should have 200 status codes'
+  );
+
+  //assert that correct custom metrics are emitted
+  t.hasProp(
+    report.aggregate.summaries,
+    'express-example.cpu',
+    "Aggregate Summaries doesn't have CPU metric"
+  );
+  t.hasProp(
+    report.aggregate.summaries,
+    'express-example.memory',
+    "Aggregate Summaries doesn't have Memory metric"
+  );
+  t.hasProp(
+    report.aggregate.histograms,
+    'express-example.cpu',
+    "Aggregate Histograms doesn't have CPU metric"
+  );
+  t.hasProp(
+    report.aggregate.histograms,
+    'express-example.memory',
+    "Aggregate Histograms doesn't have Memory metric"
+  );
+
+  //Cleanup: server
+  childProcess.kill();
+});
+
+test('cpu and memory metrics display in the aggregate report with a default name when no name is given', async (t) => {
+  //Arrange: Test Server and Plugin overrides
+  const { currentPort, childProcess, currentPid } = await startTestServer();
+  const override = JSON.stringify({
+    config: {
+      plugins: {
+        'memory-inspector': [{ pid: currentPid }]
+      }
+    }
+  });
+
+  //Act: run the test and get report
+  await $`TEST_PORT=${currentPort} ../artillery/bin/run run ./test/fixtures/scenario.yml --overrides ${override} --output ./test/output.json`;
+  const report = JSON.parse(fs.readFileSync('./test/output.json'));
+
+  //Assert
+  //sanity check that it can reach server
+  t.ok(
+    report.aggregate.counters['http.codes.200'] > 0,
+    'Should have 200 status codes'
+  );
+
+  //assert that correct custom metrics are emitted
+  t.hasProp(
+    report.aggregate.summaries,
+    `process_${currentPid}.cpu`,
+    "Aggregate Summaries doesn't have CPU metric"
+  );
+  t.hasProp(
+    report.aggregate.summaries,
+    `process_${currentPid}.memory`,
+    "Aggregate Summaries doesn't have Memory metric"
+  );
+  t.hasProp(
+    report.aggregate.histograms,
+    `process_${currentPid}.cpu`,
+    "Aggregate Histograms doesn't have CPU metric"
+  );
+  t.hasProp(
+    report.aggregate.histograms,
+    `process_${currentPid}.memory`,
+    "Aggregate Histograms doesn't have Memory metric"
+  );
+
+  //Cleanup: server
+  childProcess.kill();
+});
+
+test('cpu and memory metrics display in the aggregate report with a default name when no name is given', async (t) => {
```

Comment:
Does this test have the same name as the previous one? Is that intentional? 

If not, let's rename them to reflect what they're testing. 

Also, a large portion of the test name just repeats. We can assume that it's known that this plugin emits CPU and memory metrics, and name the tests the way that describes the tested behavior better:

```js
test('displays a default name when no name is given in the aggregate report', async () => {})
```

Reply:
copy paste blunder :). This test is about the additional metrics being present when `ARTILLERY_INTROSPECT_MEMORY` is set. I'll change the name to reflect that.

Nice catch. Thanks!!